src/js/components/message-indicator.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import DocumentTitle from 'react-document-title';
+
+const BLINKING_INTERVAL = 1500;
+
+export class MessageIndicatorComponent extends Component {
+    state = {
+        initialDocumentTitle: global.document ? document.title : '',
```

Comment:
I think before every change we should first check if the title is what we expect, something like this:

```
onTick() {
  if(title === "{name} said"){
    title = this.initialTitle;
  }else if(title == this.initialTitle){
    title = '{name} said';
  }else{
    this.initialTitle = title;
    title = '{name} said';
  }
}
```